python/pyarrow/io.pxi

```diff
@@ -121,6 +121,14 @@ cdef class NativeFile(_Weakrefable):
     def __exit__(self, exc_type, exc_value, tb):
         self.close()
 
+    def __repr__(self):
+        name = f"{self.__class__.__module__}.{self.__class__.__name__}"
+        return (f"{name}("
+                f"own_file={self.own_file}, "
+                f"is_seekable={self.is_seekable}, "
+                f"is_writable={self.is_writable}, "
+                f"is_readable={self.is_readable})")
```

Comment:
Would it be useful to add whether it is `closed` or not?

Reply:
Good idea. https://github.com/apache/arrow/pull/13921/commits/889881e40cfd5886547c0e97c5f067c986a09508